### PR TITLE
WT_SESSION.strerror

### PR DESCRIPTION
**Pull request by keithbostic**
@michaelcahill, I've kind of decided I like this change -- if you still want it, maybe Don could review & fix Java as necessary?

One important note -- I changed the extension API to match the new WT_SESSION method. I'm not sure that's a good idea or not (regardless, I'm thinking the session API isn't widely enough used that we need to worry about backward compatibility).

Reference #1516.

### CODE REVIEW

File: src/conn/conn_api.c [hunk 1]
```diff
@@ -117,7 +117,7 @@ __conn_get_extension_api(WT_CONNECTION *wt_conn)
 	conn->extension_api.conn = wt_conn;
 	conn->extension_api.err_printf = __wt_ext_err_printf;
 	conn->extension_api.msg_printf = __wt_ext_msg_printf;
-	conn->extension_api.strerror = wiredtiger_strerror;
+	conn->extension_api.strerror = __wt_ext_strerror;
```
Comment (michaelcahill) [inline src/conn/conn_api.c]: Do we still need an entry in the extension API?  i.e., are there situations where an extension will need to convert an error code to a string but won't have a session?

Comment (keithbostic) [inline src/conn/conn_api.c]: Good point -- I would certainly feel good going in that direction, at least until events prove us wrong.

Comment (keithbostic) [inline src/conn/conn_api.c]: @michaelcahill, the current Helium/Levyx driver makes a ton of calls into WiredTiger before allocating a WT_SESSION, take a look at [wiredtiger_extension_init()](https://github.com/wiredtiger/wiredtiger/blob/master/ext/datasources/helium/helium.c#L3313).

Comment (michaelcahill) [inline src/conn/conn_api.c]: OK, that's good enough for me.  I'll review properly now.

**Comment by michaelcahill** [comment]
This lgtm, but you mentioned Java issues: do you know of a problem, or just not sure that SWIG will cope with this automatically?

**Comment by keithbostic** [comment]
Just not sure that SWIG will cope automatically.

**Comment by michaelcahill** [comment]
I think SWIG will cope, happy to fix if not.  I'll merge.

**Pull request merged**